cmake/dependencies.cmake

```diff
@@ -108,6 +108,11 @@ FetchContent_Declare(
   SYSTEM
   SOURCE_SUBDIR crates/c-api)
 
+set(ABSL_PROPAGATE_CXX_STD ON)
```

Comment:
Since this is quite a foundational library, it might be worth moving this nearer the top of the file (but not above anything it depend on), due to the first declaration wins nature of fetchcontent.

Specifically, googletest depends on abseil.

Reply:
Good point. I will do a follow up PR right after this. The commit in this PR that adds the dependency is already in the tree.